src/libcore/fmt/mod.rs

```diff
@@ -265,7 +265,15 @@ pub struct ArgumentV1<'a> {
 // containing data (as a matter of static generation of the formatting
 // arguments), so this is merely an additional check.
 #[unstable(feature = "fmt_internals", reason = "internal to format_args!", issue = "none")]
-static USIZE_MARKER: fn(&usize, &mut Formatter<'_>) -> Result = |_, _| loop {};
+static USIZE_MARKER: fn(&usize, &mut Formatter<'_>) -> Result = |ptr, _| {
+    // This tries to ensure we don't collapse into some other function; we can't
+    // guarantee that doesn't happen, but this tries to make sure that if we
+    // *do* get collapsed with some other function, that function also takes a
+    // `usize` as it's first argument, ensuring correctness below when we read
+    // out the associated value from `ArgumentV1`.
+    let _v: usize = crate::hint::black_box(*ptr);
```

Comment:
Are we using `black_box` here for functional correctness?

Reply:
Yes, we were, I think I agree we probably shouldn't be. Switched to a volatile read.

Reply:
Thank you :)